synapse/handlers/sync.py

```diff
@@ -1534,12 +1534,12 @@ async def _generate_sync_entry_for_rooms(
         newly_joined_rooms = room_changes.newly_joined_rooms
         newly_left_rooms = room_changes.newly_left_rooms
 
-        async def handle_room_entries(room_entry: "RoomSyncResultBuilder"):
+        async def handle_room_entries(room_entry: "RoomSyncResultBuilder") -> None:
             with start_active_span("generate_room_entry"):
                 set_tag("room_id", room_entry.room_id)
                 log_kv({"events": len(room_entry.events or [])})
                 logger.debug("Generating room entry for %s", room_entry.room_id)
-                res = await self._generate_room_entry(
+                await self._generate_room_entry(
```

Comment:
_generate_room_entry modifies the RoomSyncResultBuilder and doesn't return anything.